docs/parameters.md

```diff
@@ -1,10 +1,53 @@
-# Using Parameters
+# Parameters
 
-Experiment parameters define the search space for assigned values that vary for each trial run. Each parameter represents a named integer assignment with an inclusive minimum and maximum bound.
+A **parameter** is a value which can be adjusted during an experiment. Every parameter has a name and a **domain** of acceptable values it can be **assigned** during a trial.
 
-## Parameter Domain
+An experiment must define one or more parameters. Together, the parameters in an experiment define the **search space**.
 
-When selecting the bounds for a parameter it is important to remember that all values are configured as integers. When tuning compute resources, such as a [CPU request](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu), you may typically use values like 0.1, 0.2, 0.3, ..., 4.0. However, for optimization you will need to specify your bounds using millicpus and add the explicit unit later, for example:
+## Parameter Types
+
+Red Sky Ops supports two types of parameters: **Integer** and **Categorical**.
```

Comment:
Is it preferable to use the RSO product name, or `redskyops-controller`, or something else?

Reply:
I think red sky ops is fine

Reply:
You do your best to be agnostic when identifying the product ("Parameters can be one of two types..."). Names change and it makes much harder to keep the documentation up to date when it's riddled with specific names.

That being said, I don't know why we are exposing "parameter types" on the user facing documentation. A parameter either has a numeric range (you specify min/max) or a string range (you specify the allowed values). It feels like this is making things more complex than they are. Maybe it's just me.

Reply:
> You do your best to be agnostic when identifying the product

My favorite solution: Avoiding the problem 😄 

---
Re. "parameter types": Honestly, I mentally organized them this way so automatically that I didn't even consider whether it was appropriate. I do think it's a helpful distinction though:

* It's consistent with how we talk about them ("We added categorical parameters in X release")
* The syntax for defining their ranges is distinct
* Certain features — parameter constraints for instance — only apply to integer parameters
* Integer ranges have certain intrinsic properties that string ranges won't (values are more meaningfully comparable, ranges are defined without explicitly defining every value)

That said, I could certainly de-emphasize the difference if you think it's best.

Reply:
I like the distinctions james has made here, especially giving the examples, I think having the use cases really helps show off when you might use each type. I like the jeremy's language of "string" better though, since I don't think the term "categorical" variable is very common and I often get asked to explain exactly what that means and why it would be useful.

Reply:
Though, I think we can drop the collection threads example with only one bound